### DISCUSSION TITLE
Kratos Quickstart Doesn't Work at v0.8.0-alpha.1

### DISCUSSION THREAD

**Discussion by designermonkey** [ory/kratos#1926] [Q&A] [answered]
John Porter Oct 26th at 5:32 PM
Hi all, it’s been a couple of years since I last asked a question here, but here I am. I don’t know what I’m doing wrong, but I cannot get the Kratos quickstart with postgres and oathkeeper to work. Navigating in the node ui to `http://127.0.0.1:4435/registration` results in a `303 See Other` which points to `http://kratos:4433/self-service/registration/browser?return_to=` which the ui obviously cannot navigate to being that it’s within the docker network and not visible outside.

Has anyone actually got this working before? If so, can anyone give me some pointers?

I’ve attempted to change the `KRATOS_PUBLIC_URL` to `http://127.0.0.1:4433/` instead of `http://kratos:4433/` which then leads to another error…

I then get a `404 Not Found` for `http://127.0.0.1:4455/auth/registration?flow=SOME_ID_HERE`.

I’m completely stumped here as no routes anywhere in the ui app are defined as prefixed with `/auth`. All of the JS and CSS also fails to load as it thinks it is behind `/auth`.

I’ve tried this without oathkeeper and I get the same 404 described above.

**Comment by aeneasr** [comment]
Welcome back! Glad to have you!

alpha.3 has resolved all of these issues! Can you try out that release?

**Comment by designermonkey** [comment]
I was trying that as you replied. Unfortunately it doesn't work:
```
// 20211103110137
// http://127.0.0.1:4455/registration?flow=83f30523-a0e2-46e8-a837-abf30702183d

{
  "error": {
    "code": 404,
    "status": "Not Found",
    "message": "Requested url does not match any rules"
  }
}
```
I navigated to `http://127.0.0.1:4435/welcome`. I then clicked on the sign up button and was eventually redirected to `http://127.0.0.1:4455/registration?flow=98b3fe83-dbd7-47eb-b308-67a3eb8efc5f` with the above error.

**Comment by vinckr** [comment]
Hey John,
thanks for the report! 

When I start the quickstart from scratch with `v0.8.0-alpha.3` checked out and
`docker-compose -f quickstart.yml -f quickstart-standalone.yml -f quickstart-postgres.yml up --build --force-recreate` everything works as expected, but it uses `http://127.0.0.1:4455/welcome`.

I am guessing that the Oathkeeper part in your setup is causing the errors. Can you share the config and ideally some steps on how I can reproduce it? And did you make changes to the quickstart.yml?

**Comment by designermonkey** [comment]
Hey,

I've pulled it all together into one yaml file and used an env file to gather shared var values, so it is customised yes, but I am being meticulous in ensuring I reproduce it properly between versions by taking the quickstart yaml's and using `docker-compose config` locally. I think it is the oathkeeper config causing the issues because if I visit `http://127.0.0.1:4455/welcome` I get:
```
/ 20211103111656
// http://127.0.0.1:4455/welcome

{
  "error": {
    "code": 404,
    "status": "Not Found",
    "message": "Requested url does not match any rules"
  }
}
```

The oathkeeper config is an exact copy of that which is in the contrib dir, so I don't know exactly how to solve the issues.

I will try the quickstart itself from kratos now with the oathkeeper config, just to be sure it's not me.

**Reply by aeneasr** [reply]
oh yeah, we have not really updated oathkeeper to the new kratos 0.8 version. can you try it without oathkeeper in the mix?

**Comment by designermonkey** [comment, marked as answer]
I can confirm it works without oathkeeper.

Problem I have is that I'm trying to evaluate this as a set up including oathkeeper. I can generally see what's going on as I've done routing like this before. Do I need to simply add `welcome` to the anonymous ruleset?

**Reply by aeneasr** [reply]
Probably! :)

**Reply by designermonkey** [reply]
Ok, so I think this is sufficient:
```
- id: 'ory:kratos-selfservice-ui-node:anonymous'
  upstream:
    preserve_host: true
    url: 'http://kratos-selfservice-ui-node:4435'
  match:
    url: 'http://127.0.0.1:4455/<{welcome,error,recovery,verification,registration,**.css,**.js,**.png}{/,}>'
    methods:
      - GET
  authenticators:
    - handler: anonymous
  authorizer:
    handler: allow
  mutators:
    - handler: noop
```

I may have missed a uri path though, but this seems to work for me now.

Thanks all for your input, it's been like rubber ducking lol

**Reply by designermonkey** [reply]
Now to figure out how to get Keto involved :D

**Reply by vinckr** [reply]
Thanks John! 
I will mark this as answered for now, thanks for sharing your fix also 🙏
